### ISSUE TITLE
Settings / Developers register new type on every keystroke 

### ISSUE THREAD

**Issue by xlc** [polkadot-js/apps#1018]
It register new types on every keystroke and don't remove the previous types.

It should
- debounce the registration to avoid action on every keystroke
- unregister previous types (not sure if it does this already)

Better than unregister previous type, it should
- save a copy of the default types
- on save button / debounce timer, revert type registry to the saved default copy, and then register new types

Does reset button actually remove registered types?

**Comment by jacogr** [comment]
I would prefer not having the editor, only the upload. 

It does not de-register, but upon reload o my the new types are there. 

Good points though, would rather dumb it down a bit with some extra upload only and have it crystal clear. Any other suggestions welcome. 

(Since you did the actual API reg, I don’t believe the dereg is needed there, or warranted - very specific case here, and can be clarified)

**Comment by xlc** [comment]
I only issue I can think about not doing deregistration is unintended registration. e.g. Register KittyDetails could unintentionally register Kitty type without user aware of it. But it is an unlikely scenario because the json is usually malformed until user entered the value part.

**Comment by xlc** [comment]
I do find editor very convenient during development. Save time compare to edit in editor and upload.

**Comment by Stefie** [comment]
I have a PR open with the following implementation:

- It still checks on every  keystroke if the input in the editor is valid JSON
- but it only registers types on explicit save, which would have made sense in the first place anyways (so no debouncing needed)
- `Reset` only removes types from the local storage, but doesn't de-register them.
- Check if the types are valid only before they are saved by moving `getTypeRegistry().register(types)` to the save function, if not, don't save and throw error

> I do find editor very convenient during development. Save time compare to edit in editor and upload.

That was exactly what I was thinking when I implemented it, but I don't have incredibly strong feelings about it.
